src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,112 @@
+
+    @Test
+    void register_userIsNull_NotOk() {
+        User newUser = null;
+        assertThrows(InvalidInputDataException.class, () ->
+                registrationService.register(newUser)
+        );
+    }
+
+    @Test
+    void register_loginIsNull_NotOk() {
+        User newUser = new User(null, VALID_PASSWORD, VALID_AGE);
+        Assertions.assertThrows(InvalidInputDataException.class,
+                () -> registrationService.register(newUser), "Login can't be null"
+        );
+    }
+
+    @Test
+    void register_userWithSuchLoginAlreadyExist_NotOk() {
+        User newUser = new User(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertThrows(InvalidInputDataException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void register_userAgeLessThanMin_NotOk() {
+        User newUser = new User(DEFAULT_LOGIN, VALID_PASSWORD, VALID_AGE - 5);
+        Assertions.assertThrows(RuntimeException.class,
+                () -> registrationService.register(newUser), "Not valid age"
+        );
+    }
+
+    @Test
+    void register_userAgeEnough_Ok() {
+        User user = new User(DEFAULT_LOGIN, VALID_PASSWORD, VALID_AGE + 5);
+        assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    public void register_ageFloorEdge_NotOk() {
+        User newUser = new User(DEFAULT_LOGIN, VALID_PASSWORD, VALID_AGE - 1);
+        assertThrows(InvalidInputDataException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    public void register_userAgeIsNull_NotOk() {
+        User user = new User(DEFAULT_LOGIN, VALID_PASSWORD, null);
+        Assertions.assertThrows(RuntimeException.class, () ->
+                registrationService.register(user), "Age cant be null");
+    }
+
+    @Test
+    void register_passwordIsNull_NotOk() {
+        User newUser = new User(DEFAULT_LOGIN, null, VALID_AGE);
+        Assertions.assertThrows(RuntimeException.class,
+                () -> registrationService.register(newUser)
+        );
+    }
+
+    @Test
+    void register_checkPasswordLength_NotOk() {
+        User newUser = new User(DEFAULT_LOGIN, "abc", VALID_AGE);
+        Assertions.assertThrows(RuntimeException.class,
+                () -> registrationService.register(newUser)
+        );
+    }
+
+    @Test
+    void register_passwordLengthOk_Ok() {
+        User newUser = new User(DEFAULT_LOGIN, "enoughLengthPassword", VALID_AGE);
+        assertEquals(newUser, registrationService.register(newUser));
+    }
+
+    @BeforeAll
+    public static void init() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        User firstUser = new User(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        User secondUser = new User(SECOND_VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
```

Comment:
You don't need to create 2 users every time when test calls, 1 is more than enough